File: Practica 1/C++/combinaciones_2.cpp
#include<bits/stdc++.h>
using namespace std;

void imprimirCombi(string cadena, int n, int tam, ofstream &myfile){
  if(tam == n){
    myfile << cadena << ',';
    return;
  }
  string cadena_0 = cadena+"0";
  string cadena_1 = cadena+"1";
  imprimirCombi(cadena_0,n,tam+1,myfile);
  imprimirCombi(cadena_1,n,tam+1,myfile);
}

int main(){
  ofstream myfile;
  myfile.open ("datos2.txt");
  myfile << '{';
  int opc,n;
  string temp = "";
  cout << "Ingrese n: ";
  cin >> n;
  for(int i = 1; i < n+1; i++){
    imprimirCombi(temp,i,0, myfile);
    cout <<  i << '\n';
  }
  myfile<<'}';
  cout << "Archivo generado\n";
  myfile.close();
}

File: Practica 1/python/combinaciones_2.py
import matplotlib.pyplot as plt
import numpy as np
n = int(input("Ingrese una n: "))

file = open("datos_2.txt", "w")
file.write('{E')

cadena = ''

unos = []

def guardarCadenas(cadena, tam,n):
  if tam == n:
    file.write(cadena + ',')
    unos.append(cadena.count("1"))
    return
  cadena_0 = cadena+'0'
  cadena_1 = cadena+'1'
  guardarCadenas(cadena_0,tam+1,n)
  guardarCadenas(cadena_1,tam+1,n)
for i in range(n+1):
  guardarCadenas(cadena,0,i)
file.write("}")
file.close()
print("Cadena generada")
plt.scatter(np.array(range(len(unos))), unos,  s=10)
plt.xlabel("Numero de Cadena")
plt.ylabel("Numero de 1's")
unos.clear()
plt.show()

